src/python/pants/backend/python/rules/python_test_runner_integration_test.py

```diff
@@ -1,266 +1,376 @@
-        platform SOME_TEXT
-        rootdir: SOME_TEXT
-        plugins: SOME_TEXT
-        collected 1 item
 
-        pants/dummies/test_with_source_dep_absolute_import.py .                  [100%]
-
-        ============================== 1 passed in SOME_TEXT ===============================
-
-
-        testprojects/tests/python/pants/dummies:target_with_source_dep_absolute_import  .....   SUCCESS
-        """)
+  def run_pytest(self, *, passthrough_args: Optional[str] = None) -> TestResult:
+    target = PythonTestsAdaptor(
+      address=BuildFileAddress(rel_path=f"{self.source_root}/BUILD", target_name="target"),
+    )
+    pytest_subsystem = global_subsystem_instance(
+      PyTest, options={
+        PyTest.options_scope: {
+          "args": [passthrough_args] if passthrough_args else [],
+          "version": "pytest>=4.6.6,<4.7",  # so that we can run Python 2 tests
+        },
+      },
+    )
+    test_target_setup = self.request_single_product(
+      TestTargetSetup,
+      Params(
+        target,
+        pytest_subsystem,
+        PythonNativeCode.global_instance(),
+        PythonSetup.global_instance(),
+        SourceRootConfig.global_instance(),
+        SubprocessEnvironment.global_instance(),
+      ),
+    )
+    # TODO: replace all this boilerplate with a utility to set up OptionsBootstrapper for
+    # non-ConsoleRuleTestBase tests.
+    mock_global_options = OptionValueContainer()
+    mock_global_options.colors = RankedValue(RankedValue.HARDCODED, False)
+    test_result = self.request_single_product(
+      TestResult,
+      Params(
+        target,
+        test_target_setup,
+        PythonSetup.global_instance(),
+        SubprocessEnvironment.global_instance(),
+        GlobalOptions(mock_global_options),
       )
-
-  def test_source_dep_relative_import(self):
-    pants_run = self.run_passing_pants_test([
-      'testprojects/tests/python/pants/dummies:target_with_source_dep_relative_import',
-    ])
-    self.assert_fuzzy_string_match(
-      pants_run.stdout_data,
-      dedent("""\
-        testprojects/tests/python/pants/dummies:target_with_source_dep_relative_import stdout:
-        ============================= test session starts ==============================
-        platform SOME_TEXT
-        rootdir: SOME_TEXT
-        plugins: SOME_TEXT
-        collected 1 item
-
-        pants/dummies/test_with_source_dep_relative_import.py .                  [100%]
-
-        ============================== 1 passed in SOME_TEXT ===============================
-
-
-        testprojects/tests/python/pants/dummies:target_with_source_dep_relative_import  .....   SUCCESS
-        """)
+    )
+    debug_result = self.request_single_product(
+      TestDebugResult, Params(test_target_setup, InteractiveRunner(self.scheduler)),
+    )
+    if test_result.status == Status.SUCCESS:
+      assert debug_result.exit_code == 0
+    else:
+      assert debug_result.exit_code != 0
```

Comment:
This is all it takes to ensure the `debug_python_test` works for every single test below! Yay V2!